### DISCUSSION TITLE
Adding Basic Auth to allow access to secured URL's

### DISCUSSION THREAD

**Discussion by EricMeeks** [mitre/HTTP-Proxy-Servlet#201] [Ideas]
Want to use this to proxy to a SOLR instance that is protected by basis auth. Would like it so that some configured username/password set of credentials is added every time I proxy into SOLR. Can probably add this myself, but is this a feature that others might want? 

Note that I will be protecting this Servlet in other ways (Shibboleth or LDAP stuff) so I'm not going to be exposing a locked down SOLR instance in an unprotected way. Really, I'm just trying to use this to add more security flexibility into SOLR.

**Comment by dsmiley** [comment]
Sounds useful!  PR welcome.

**Comment by EricMeeks** [comment]
Hello again.
Just want to write to thank you and let you know that we did modify your code to allow various basic auth users to access the proxy, which then uses a single system level basic auth user to access a protected API (Apache SOLR in our case). It was only a few tweaks of code to make this happen.

This is the product https://project-emerse.org/

We’ve only done this for our UCSF version of the code, but we are giving a presentation on Thursday to the EMERSE community where we will talk about a bunch of things including the use of your proxy to manage SOLR access in a HIPAA compliant manner. I expect some others will want this as well at some point.

Thanks!
Eric
